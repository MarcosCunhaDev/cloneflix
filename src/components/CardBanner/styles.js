import styled from "styled-components/native";
import { colors } from "../../config/const";

export const BoxCard = styled.SafeAreaView`
  width: 200px;
  height: 250px;
  justify-content: flex-start;
  align-items: center;
  margin: 10px 10px;
  margin-left: 0px;
  overflow: hidden;
  border-radius: 5px;
`;

export const ContentImage = styled.View`
  width: 200px;
  height: 250px;
  border-radius: 5px;
  overflow: hidden;
`;
