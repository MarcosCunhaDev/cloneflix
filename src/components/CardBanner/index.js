import React from "react";
import { View, Text, Image } from "react-native";
import { BoxCard, Title, ContentImage, BoxTitle } from "./styles";
import { IMAGE_BASE_URL, IMAGE_SIZE } from "../../services/api";

const Card = ({ title, backPath }) => {
  return (
    <BoxCard>
      <ContentImage>
        <Image
          source={{
            uri:
              `${IMAGE_BASE_URL}${IMAGE_SIZE}${backPath}` ||
              "https://upload.wikimedia.org/wikipedia/pt/f/f5/Marvels_avengers_capa.jpg",
          }}
          style={{ flex: 1, resizeMode: "stretch" }}
        />
      </ContentImage>
    </BoxCard>
  );
};

export default Card;
