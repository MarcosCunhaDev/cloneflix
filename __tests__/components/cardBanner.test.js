import React from "react";
import renderer from "react-test-renderer";

import CardBanner from "../../src/components/CardBanner";

describe("<CardBanner />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<CardBanner />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});

it("renders correctly", () => {
  const tree = renderer.create(<CardBanner />).toJSON();
  expect(tree).toMatchSnapshot();
});
