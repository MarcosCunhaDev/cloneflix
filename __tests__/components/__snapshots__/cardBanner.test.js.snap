// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "height": 250,
        "justifyContent": "flex-start",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 10,
        "marginTop": 10,
        "overflow": "hidden",
        "width": 200,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 5,
          "height": 250,
          "overflow": "hidden",
          "width": 200,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://image.tmdb.org/t/p/originalundefined",
        }
      }
      style={
        Object {
          "flex": 1,
          "resizeMode": "stretch",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
